### PR TITLE
Update helper.py to fix #8785

### LINKED ISSUE #8785: Jira assignee not set at issue creation time causes errors
**Be informative**
Since Jira assignee is not set at create_issue API call but is actually edited after with "default_assignee" set at Jira project settings, some issues creation will fail. This happens when the Jira project doesn't allow empty or 'unusigned' users to be set.

**Bug description**
Jira tickets can't be created under certain circumstances (explained above).
Also, creating and then editing a jira ticket causes 2 requests, when only 1 is actually necessary. (PR to follow to fix it)

**Steps to reproduce**
1. set a default assignee at jira project settings

**Expected behavior**
When fixed, jira issues will be created with the default assignee instead of creating and then editing it.

**Deployment method** *(select with an `X`)*
- [ x] Docker Compose
- [ ] Kubernetes
- [ ] GoDojo

**Environment information**
 - Linux
 - Defect Dojo 2.23.2 (it applies to any version)

**Logs** 
NA

**Additional context** (optional)
will create a PR that fixes this issue and do both tasks in just one request (API call).

### PR DESCRIPTION
**Pull request by devsecopsale**
this will fix #8785

## :warning: Note on feature completeness :warning:

We are narrowing the scope of acceptable enhancements to DefectDojo in preparation for v3. Learn more here:
https://github.com/DefectDojo/django-DefectDojo/blob/master/readme-docs/CONTRIBUTING.md

**Description**

this will fix #8785 , by adding the jira.project default_assignee (when defined) to the issue creation api call instead of adding a second step to do it.

**Test results**

this was tested in 2.23.2 and allowed to create issues in a Jira project where no 'unassigned' or 'automatic' assignee was allowed.

**Documentation**

**Checklist**

This checklist is for your information.

- [ ] Make sure to rebase your PR against the very latest `dev`.
- [ ] Features/Changes should be submitted against the `dev`.
- [ ] Bugfixes should be submitted against the `bugfix` branch.
- [ ] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [ ] Your code is flake8 compliant.
- [ ] Your code is python 3.11 compliant.
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [ ] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ ] Add applicable tests to the unit tests.
- [ ] Add the proper label to categorize your PR.

**Extra information**

Please clear everything below when submitting your pull request, it's here purely for your information.

Moderators: Labels currently accepted for PRs:
- Import Scans (for new scanners/importers)
- enhancement
- performance
- feature
- bugfix
- maintenance (a.k.a chores)
- dependencies
- New Migration (when the PR introduces a DB migration)
- settings_changes (when the PR introduces changes or new settings in settings.dist.py)

# Contributors: Git Tips
## Rebase on dev branch
If the dev branch has changed since you started working on it, please rebase your work after the current dev.

On your working branch `mybranch`:
```
git rebase dev mybranch
```
In case of conflict:
```
 git mergetool
 git rebase --continue
 ```

When everything's fine on your local branch, force push to your `myOrigin` remote:
```
git push myOrigin --force-with-lease
```

To cancel everything:
```
git rebase --abort
```

## Squashing commits
```
git rebase -i origin/dev
```
- Replace `pick` by `fixup` on the commits you want squashed out
- Replace `pick` by `reword` on the first commit if you want to change the commit message
- Save the file and quit your editor

Force push to your `myOrigin` remote:
```
git push myOrigin --force-with-lease
```

### CODE REVIEW

File: dojo/jira_link/helper.py [hunk 1]
```diff
@@ -745,13 +749,12 @@ def add_jira_issue(obj, *args, **kwargs):
...
             duedate=duedate,
-            issuetype_fields=issuetype_fields)
+            issuetype_fields=issuetype_fields,
+            default_assignee=jira_project.default_assignee)
 
         logger.debug('sending fields to JIRA: %s', fields)
         new_issue = jira.create_issue(fields)
-        if jira_project.default_assignee:
-            jira.assign_issue(new_issue.key, jira_project.default_assignee)
-
```
Comment (Maffooch) [inline dojo/jira_link/helper.py:754]: I see why you've done this here, but adding an additional check on the nee_issue to check if the assignee was set correctly would be even better. This way, there would have to be two failures that would need to occur for a jira issue to not have an assignee

**Review by Maffooch — CHANGES_REQUESTED** [review]

File: dojo/jira_link/helper.py (hunk 1 above)
Comment (devsecopsale) [reply dojo/jira_link/helper.py:754]: You mean to check for example if default_assignee is not empty? or check if the user exists?

Comment (Maffooch) [reply dojo/jira_link/helper.py:754]: I mean to check if the jira issue has an assignee. If not, then the manual step of assigning a user can be used as a backup

Comment (devsecopsale) [reply dojo/jira_link/helper.py:754]: the proposed change only affects the use case when a ticket doesn't exist yet (add_jira_issue() function). The assignee is set for the new ticket at the moment of creation only when a 'default assignee' has been set in the Jira Project setting (at engagement or product level jira settings - depending on inheritance option). If no jira default assignee is set, then the current behaviour doesn't change and the ticket is still attempted to be created with no assignee.

Comment (Maffooch) [reply dojo/jira_link/helper.py:754]: Apologies, I do not think I am doing a good job of making this clear. Here is a the flow I am trying to convey:
- User pushes ticket to Jira
- Default assignee is inserted into the options of the jira ticket creation function (the change you are introducing)
- After the ticket is created, check if the default assignee is indeed associated with the ticket (something new I am requesting)
- If the previous step is not true, then run the manual step of assigning the default assigned to the jira ticket (leaving the current behavior as a backup)
The extra step will hopefully not be called all that often, but is nice to have as a backup if the scenario were to arise

Please let me know If I can explain in another way

Comment (devsecopsale) [reply dojo/jira_link/helper.py:754]: got it now. Thank you for the clarification. I'll add the check

Comment (Maffooch) [reply dojo/jira_link/helper.py]: This is exactly what I was looking for! Thank you :)

**Comment by devsecopsale** [comment]
> Flake8 is very unhappy :/ please take a look

For what I could see, none of the warnings come from the changes I made.

**Comment by devsecopsale** [comment]
> > Flake8 is very unhappy :/ please take a look
> 
> For what I could see, none of the warnings come from the changes I made.

hi @Maffooch , just checking if you had the chance to read my last response. thanks

**Comment by devsecopsale** [comment]
Hi @Maffooch , I fixed flake8 errors too. There's now an error reported by git_protect.py. Not a clue why this is failing.

Run python .github/scripts/git_protect.py 1c8e375f2387ef263982127a67befa6e3f308c81 86e5d68b17a62ce07a69e21379542b9ff2b69e9b --comment-only &> output.txt
Error: Process completed with exit code 2.

**Comment by devsecopsale** [comment]
hi @Maffooch , any idea of what this check is about? can this be rerun?

**Comment by mtesauro** [comment]
@devsecopsale About that test - see here: https://github.com/DefectDojo/django-DefectDojo/discussions/8974

It allows us to signal contributors that they are touching files that we may be actively working on updating/doing a re-write. I don't have time right now to review this PR but that test isn't something you can "pass" short of removing the file(s) from the PR which probably isn't possible for this one.

Think of this as a heads-up that we're going to look a bit more closely at this PR vs a parser PR for example.

**Review by Maffooch — APPROVED** [review]

**Comment by devsecopsale** [comment]
@dougmorato , @Juu: I just need 2 more approvals to merge. Thank you

**Review by grendel513 — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Pull request merged**